mono/mini/mini.c

```diff
@@ -4123,7 +4123,7 @@ mono_jit_compile_method_inner (MonoMethod *method, MonoDomain *target_domain, in
 		shared = NULL;
 	}
 
-	mono_domain_lock (target_domain);
+	mono_stats_lock ();
```

Comment:
I believe we still need the domain lock here for other things. You should just take the runtime stats lock around the stats-relevant code.